bugwarrior/services/logseq.py

```diff
@@ -0,0 +1,301 @@
+            )
+            return self.json_response(response)
+        except requests.exceptions.ConnectionError as ce:
+            log.fatal("Unable to connect to Logseq HTTP APIs server. %s", ce)
+            exit()
+
+    def get_graph_name(self):
+        graph = self._get_current_graph()
+        return graph["name"] if graph else None
+
+    def get_issues(self):
+        task_filter = self.filter.replace(" ", "").replace(",", '" "')
+        return self._datascript_query(
+            f"""
+            [:find (pull ?b [*])
+                :where [?b :block/marker ?marker]
+                [(contains? #{{\"{task_filter}\"}} ?marker)]
+            ]
+        """
+        )
+
+
+class LogseqIssue(Issue):
+    ID = "logseqid"
+    UUID = "logsequuid"
+    STATE = "logseqstate"
+    TITLE = "logseqtitle"
+    SCHEDULED = "logseqscheduled"
+    DEADLINE = "logseqdeadline"
+    DONE = "logseqdone"
+    URI = "logsequri"
+
+    # Local 2038-01-18, with time 00:00:00.
+    # A date far away, with semantically meaningful to GTD users.
+    # see https://taskwarrior.org/docs/dates/
+    SOMEDAY = datetime(2038, 1, 18)
+
+    UDAS = {
+        ID: {
+            "type": "string",
+            "label": "Logseq ID",
+        },
+        UUID: {
+            "type": "string",
+            "label": "Logseq UUID",
+        },
+        STATE: {
+            "type": "string",
+            "label": "Logseq State",
+        },
+        TITLE: {
+            "type": "string",
+            "label": "Logseq Title",
+        },
+        SCHEDULED: {
+            "type": "date",
+            "label": "Logseq Scheduled",
+        },
+        DEADLINE: {
+            "type": "date",
+            "label": "Logseq Deadline",
+        },
+        DONE: {
+            "type": "date",
+            "label": "Logseq Done",
+        },
+        URI: {
+            "type": "string",
+            "label": "Logseq URI",
+        },
+    }
+
+    UNIQUE_KEY = (ID, UUID)
+
+    # map A B C priority to H M L
+    PRIORITY_MAP = {
+        "A": "H",
+        "B": "M",
+        "C": "L",
+    }
+
+    STATE_MAP = {
+        "IN-PROGRESS": "pending",
+        "DOING": "pending",
+        "TODO": "pending",
+        "NOW": "pending",
+        "LATER": "waiting",
+        "WAIT": "waiting",
+        "WAITING": "waiting",
+        "DONE": "completed",
+        "CANCELED": "deleted",
+        "CANCELLED": "deleted",
+    }
+
+    # replace characters that cause escaping issues like [] and "
+    def _unescape_content(self, content):
+        return (
+            content.replace('"', "'")  # prevent &dquote; in task details
+            .replace("[[", self.config.char_open_link)  # alternate brackets for linked items
+            .replace("]]", self.config.char_close_link)
+            .replace("[", self.config.char_open_bracket)  # prevent &open; and &close;
+            .replace("]", self.config.char_close_bracket)
+        )
+
+    # get a optimized and
```

Comment:
?

Reply:
This is a work around for what is possibly an underlying bug in taskwarrior. The `[` and `]` characters in the task description, and other fields, are replaced with `&open;` and `&close;`.

Reply:
I'm not sure this is the review comment you meant to respond to. This comment was about the python comment above which seems like an incomplete thought.

Reply:
LOL. Yes, an incomplete thought. I think I meant something like "get an optimized title"